subsys/usb/class/mass_storage.c

```diff
@@ -290,74 +292,98 @@ static void testUnitReady(void)
 	sendCSW();
 }
 
-static bool requestSense(void)
+static struct fixed_format_sense_data sense_data = {
+	.code = SDRC_CURRENT_ERRORS,
+	.sense_key = SK_ILLEGAL_REQUEST,
+	.as_length = sizeof(struct additional_sense_data),
+	.asd = {
+		.asc_ascq = {0x30, 0x01},
+	},
+};
+
+static void update_sense_data_ascq(u16_t ascq)
+{
+	sys_put_be16(ascq, sense_data.asd.asc_ascq);
+}
+
+static bool req_sense_cmd(void)
 {
-	u8_t request_sense[] = {
-		0x70,
-		0x00,
-		0x05,   /* Sense Key: illegal request */
-		0x00,
-		0x00,
-		0x00,
-		0x00,
-		0x0A,
-		0x00,
-		0x00,
-		0x00,
-		0x00,
-		0x30,
-		0x01,
-		0x00,
-		0x00,
-		0x00,
-		0x00,
-	};
-
-	return write(request_sense, sizeof(request_sense));
+	struct cdb6 *cmd = (struct cdb6 *)&cbw.CB[0];
+
+	return write((u8_t *)&sense_data,
+		     MIN(cdb6_get_length(cmd), sizeof(sense_data)));
 }
 
-static bool inquiryRequest(void)
+static struct dabc_inquiry_data inq_data = {
+	.type = DIRECT_ACCESS_BLOCK_DEVICE,
+	.qualifier = 0,
+	.rmb = 1,	/* Medium is not removable */
+	.version = 0,
+	.rdf = 2,
+	.length = sizeof(struct dabc_inquiry_data) - 5,
```

Comment:
Specification strictly mentions  ```0x1F``` so maybe it is worthwhile to make it hard coded and underline the need?